distributed-job-manager/src/test/java/ru/fix/distributed/job/manager/strategy/RendezvousHashAssignmentStrategyTest.java

```diff
@@ -0,0 +1,168 @@
+package ru.fix.distributed.job.manager.strategy;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import ru.fix.distributed.job.manager.model.AssignmentState;
+import ru.fix.distributed.job.manager.model.JobId;
+import ru.fix.distributed.job.manager.model.WorkItem;
+import ru.fix.distributed.job.manager.model.WorkerId;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static ru.fix.distributed.job.manager.strategy.AssignmentStrategyUtils.*;
+
+class RendezvousHashAssignmentStrategyTest {
+    private RendezvousHashAssignmentStrategy rendezvous;
+
+    @BeforeEach
+    void setUp() {
+        rendezvous = new RendezvousHashAssignmentStrategy();
+    }
+
```

Comment:
composition of workItems in java is hard to read. Suggestion: lets add print function and log assignment process in the test. This way reassignment scenario and behavior will be clear.  E.g.
`fun  print(availability, prevAssign, itemsToAssing, newAssignmentState)`
that will print
```
available:
    └ job-0
        └ worker-0
    └ job-1
        └ worker-0
    └ job-2
        └ worker-0
    └ job-3
        └ worker-1
prev: Assignment state
	└ worker-2
	└ worker-1
	└ worker-0
items:
    └ job-0
        └ work-item-2
        └ work-item-1
    └ job-1   
      └ work-item-2
       ...
new: Assignment state
	└ worker-1
		└ Job[job-3] - work-item-2
		└ Job[job-3] - work-item-1
		└ Job[job-3] - work-item-0
	└ worker-0
		└ Job[job-0] - work-item-2
		└ Job[job-0] - work-item-1
		└ Job[job-1] - work-item-2
		└ Job[job-0] - work-item-0
		└ Job[job-2] - work-item-2
		└ Job[job-1] - work-item-1
		└ Job[job-2] - work-item-1
		└ Job[job-1] - work-item-0
		└ Job[job-2] - work-item-0
```
This printing function can be shared between both tests: and for randezvous and evenly spread tests. 
Also this printing function can be used in production code where you are reporting assignment process in the trace logs. 
Without this is is very hard to spot an issue.

Reply:
Done